sway/xdg_decoration.c

```diff
@@ -23,8 +23,28 @@ static void xdg_decoration_handle_request_mode(struct wl_listener *listener,
 		void *data) {
 	struct sway_xdg_decoration *deco =
 		wl_container_of(listener, deco, request_mode);
+	struct sway_view *view = deco->view;
+	bool wants_floating = view->impl->wants_floating;
```

Comment:
`impl->wants_floating` is a function pointer, so need to call the function to get the `bool`. The function may be NULL.

If we have a container attached to the view, it's better to use `container_is_floating` to find out whether the view is floating. If we don't have a container (e.g. toplevel isn't mapped yet), using `wants_floating` sounds reasonable.